python/ray/autoscaler/_private/monitor.py

```diff
@@ -155,6 +155,8 @@ def __init__(self,
             gcs_address, options)
         self.gcs_node_resources_stub = \
             gcs_service_pb2_grpc.NodeResourceInfoGcsServiceStub(gcs_channel)
+        self.gcs_node_info_stub = \
```

Comment:
Do we need to define it here? Why not do it inside autoscaler.p?

Reply:
To share the channel `gcs_channel`, the code required to instantiate the channel (lines 143 through 155), and the data required to instantiate (redis password and redis address).

I would actually feel a bit better if the drain rpc call were made from monitor.py and autoscaler.py didn't know anything about RPCs, but that would require more significant refactors.